internal/river_collector/openbmc.go

```diff
@@ -0,0 +1,204 @@
+// MIT License
+//
+// (C) Copyright [2020-2021,2024] Hewlett Packard Enterprise Development LP
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+// OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+package river_collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Cray-HPE/hms-hmcollector/internal/hmcollector"
+	rf "github.com/Cray-HPE/hms-smd/pkg/redfish"
+)
+
+func (collector OpenBMCRiverCollector) ParseJSONPowerEvents(payloadBytes []byte,
+	location string) (events []hmcollector.Event) {
+	timestamp := time.Now().Format(time.RFC3339)
+
+	var power hmcollector.Power
+	decodeErr := json.Unmarshal(payloadBytes, &power)
+	if decodeErr != nil {
+		return
+	}
+
+	// PowerControl
+	powerControlEvent := hmcollector.Event{
+		MessageId:      PowerMessageID,
+		EventTimestamp: time.Now().Format(time.RFC3339),
```

Comment:
Should this reference timestamp like the other events?

Reply:
Yes, thanks for finding that.